services/puppetforge/puppetforge-module-pdk-version.service.js

```diff
@@ -0,0 +1,46 @@
+'use strict'
+
+const { NotFound } = require('..')
+const { renderVersionBadge } = require('../version')
+const { BasePuppetForgeModulesService } = require('./puppetforge-base')
+
+module.exports = class PuppetforgeModulePdkVersion extends BasePuppetForgeModulesService {
+  static get category() {
+    return 'platform-support'
+  }
+
+  static get route() {
+    return {
+      base: 'puppetforge/pdk-version',
+      pattern: ':user/:moduleName',
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Puppet Forge – PDK version',
+        namedParams: {
+          user: 'tragiccode',
+          moduleName: 'azure_key_vault',
+        },
+        staticPreview: renderVersionBadge({ version: '1.7.1' }),
+      },
+    ]
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'pdk version' }
+  }
+
+  async handle({ user, moduleName }) {
+    const data = await this.fetch({ user, moduleName })
+    if (data.current_release.pdk) {
```

Comment:
is there anyway `pdk` could be true, but the `data.current_release.metadata` object key could be undefined? I'm not familiar with the service/api response but am assuming there's a relationship between those two keys based on the schema and this logic?

Reply:
Not sure, but I've tightened up the schema a bit so that if that condition does happen, the validation will fail